### ISSUE TITLE
Missing gir1.2-appindicator3-0.1 dependency

### ISSUE THREAD

**Issue by hotice** [almostearthling/when-command#47]
The when-command deb doesn't depend on the "gir1.2-appindicator3-0.1" package and for this reason, it fails to work:
```
$ /opt/when-command/when-command
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 2135, in _find_spec
AttributeError: 'DynamicImporter' object has no attribute 'find_spec'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/when-command/when-command", line 51, in <module>
    from gi.repository import AppIndicator3 as AppIndicator
  File "/usr/lib/python3/dist-packages/gi/importer.py", line 53, in find_module
    'introspection typelib not found' % namespace)
ImportError: cannot import name AppIndicator3, introspection typelib not found
```
Installing "gir1.2-appindicator3-0.1" fixes this issue (while this package may be installed by default in Ubuntu, it's not in say Linux Mint Cinnamon, etc.).

**Comment by almostearthling** [comment]
Thank you for submitting this: I'll fix it for the actual way of packaging **When** (and definitely in the next way, which per #44 should follow the LHS). Lintian also reported that I should include, just for one (grin), the Python interpreter itself... I'm just learning how to package and discovered that I give too many things for granted.

Are you trying the applet on Mint? I'm going to be OT, but it would be useful to test event conditions that require user interaction (especially the screen lock based one, which relies on events of handled via the `com.ubuntu.Upstart0_6` interface).

**Comment by hotice** [comment]
I'm testing it because I'm writing an article about When on http://www.webupd8.org and I usually test the apps I write about in multiple Ubuntu versions as well as derivatives / distros based on it.

I just took a look at what you said in Linux Mint 17.2 (VirtualBox) and the screensaver as well as the session lock/unlock events are disabled.

**Comment by almostearthling** [comment]
I'll have to find a way to define such events in a more parametric way, maybe reusing the generic DBus interface, or using a separate module for events. Thank you for your interest in When, and for helping with these quirks!

**Comment by hotice** [comment]
By the way, if you need help with the debian/ubuntu packaging, let me know how I can help.

**Comment by almostearthling** [comment]
I will for sure: I've been using your PPA for several applications, so things that I still find difficult to grasp might be a breeze for you... :clap: I started a new branch for packaging, but I'm still in the reorganization phase.

**Comment by hotice** [comment]
I can only help with the debianization. Ideally everything needs to installed via the makefile or setup.py and I don't have experience with that, but the packaging can also simply tell where each file should go when installed (but that's more messy). 

Handling everything in the packaging directly can already be done and if that's the way you want it done, I can get it ready for a PPA already, just let me know. Or, I can wait until you add a setup.py with everything and package it then.

**Comment by almostearthling** [comment]
I'm actually aiming at a makefile-based way of determining file destinations: installing in `/opt` is comfortable (and lazy) but it's not standard; tools like lintian complain about it and it's reasonable, because stuff that goes to `/opt` shouldn't be handled by packages. However, as a pre-release and in order to keep things as clean as possible, it seems a fairly good choice for now: I'll update the *.deb* ASAP, probably in the following hours, in order to close the issue within days. Meanwhile I'm working on the makefile that would allow for FHS compliance, which will of course include the missing dependencies you noticed and maybe other ones.

**Comment by almostearthling** [comment]
BTW, thanks for the article! I hope that with an user base some bugs that are not tested (or cannot be tested automatically) will come to attention, so that I can make the applet more robust!

**Comment by almostearthling** [comment]
For now I added the dependency to the old version of the control file, and updated the release with a new `.deb`. However I'd leave this issue open until the new package system is ready if it's ok.

**Comment by hotice** [comment]
Thank you!

**Comment by almostearthling** [comment]
Hi @hotice, I finally managed to find a way to produce more LSB-compliant packages. Everythong is included in the [packaging](https://github.com/almostearthling/when-command/tree/packaging) branch, which probably doesn't require much more work before I "self-PR". I automated the packaging process into a script that does everything that is suggested in `PACKAGE.md`. I'll try to see if the obtained source package files and `.deb` are suitable for Launchpad, but in case of need I'll ask for your help.

Thanks for your support!
